BasicObstacleAvoidence/BasicObstacleAvoidence.ino

```diff
@@ -0,0 +1,63 @@
+#include <Smartcar.h>
+
+const unsigned short LEFT_ODOMETER_PIN = 2;
+const unsigned short RIGHT_ODOMETER_PIN = 3;
+const unsigned long PRINT_INTERVAL = 100;
+unsigned long previousPrintout = 0;
+const int TRIGGER_PIN = 6; //D6
+const int ECHO_PIN = 7; //D7
+const unsigned int MAX_DISTANCE = 100;
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+
+BrushedMotor leftMotor(8, 10, 9);
+BrushedMotor rightMotor(12, 13, 11);
+DifferentialControl control(leftMotor, rightMotor);
+
+GY50 gyroscope(37);
+DirectionlessOdometer leftOdometer(100);
+DirectionlessOdometer rightOdometer(100);
+
+SmartCar car(control, gyroscope, leftOdometer, rightOdometer);
+
+void setup() {
+  Serial.begin(9600);
+  // Initialize the odometers (they won't work otherwise)
+  leftOdometer.attach(LEFT_ODOMETER_PIN, []() {
+    leftOdometer.update();
+  });
+  rightOdometer.attach(RIGHT_ODOMETER_PIN, []() {
+    rightOdometer.update();
+  });
+
+  car.enableCruiseControl();
+  car.setSpeed(1.5); // Maintain a speed of 1.5 m/sec
+}
+
+void loop() {
+  // Maintain the speed and update the heading
+  car.update();
+  int colli = front.getDistance();        //distance measured by the ultrasonic sensors
+  Serial.println("obstacle at  : ", colli);
+
+  delay(100);
+  car.getHeading();
+  car.getDistance();
+  
+    if (colli<20 && colli>0){
+      car.setSpeed(0);        //stop moving car
+    }  
+    else{
+      car.setSpeed(1.5);
+      car.update();
```

Comment:
You don't need this extra `car.update()` :alien:

Reply:
i didnt have it in the beginnign but i read somewhere while going through the method update that it is good to update it very often. :) But will be removing it :)

Reply:
Yes, it's good to call `update()` as often as possible, however since you already do  that in the `loop` it doesn't make much difference if you call it there as well. :grin: